### PR TITLE
ARROW-5420: [Java] Implement or remove getCurrentSizeInBytes in Variab…

### PR DESCRIPTION
**Pull request by tianchen92**
Related to [ARROW-5420](https://issues.apache.org/jira/browse/ARROW-5420).

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java [hunk 1]
```diff
@@ -571,8 +571,10 @@ public int getByteCapacity() {
 
   @Override
   public int getCurrentSizeInBytes() {
-    /* TODO */
-    return 0;
+    if (valueCount == 0) {
+      return 0;
+    }
+    return offsetBuffer.getInt(valueCount * OFFSET_WIDTH);
```
Comment (emkornfield) [inline java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java]: I think you should probably include valueBuffer somehow as well?

**Review by emkornfield — CHANGES_REQUESTED** [review]
Thank you @tianchen92 could you add a unit test?  Also if the build is failing try rebasing.

File: java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java (hunk 1 above)
Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java]: Thanks for your comments, I will add a test later. 
And what do you mean "includeBuffer"? It seems getCurrentSizeInBytes means the valueBuffer size in Drill.

Comment (emkornfield) [reply java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java]: I was thinking it should be offset buffer + size of valueBuffer (I think the last value in offsetBuffer) as that would be the total size but I guess it isn't well defined (no comments on intent on the javadoc)

Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java]: Not sure, you mean the implementation like this？
`final int offsetBufferSize = (valueCount + 1) * OFFSET_WIDTH;`
 `final int dataBufferSize = offsetBuffer.getInt(valueCount * OFFSET_WIDTH);`
 ` return offsetBufferSize + dataBufferSize;`

if so, why not include validityBuffer, then this method is duplicated with getBufferSizeFor.
And this method in drill : https://github.com/apache/drill/blob/master/exec/vector/src/main/codegen/templates/VariableLengthVectors.java#L125

Comment (emkornfield) [reply java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java]: Thank you for the pointer.  It seems reasonable to mimic what they are doing?  Could you document on the interface what will be returned?

Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java]: Sure~

**Comment by tianchen92** [comment]
@emkornfield I add a unit test and document, please take a look, thanks!
And after rebasing code, build still fails with python test(have tried several times).

**Comment by tianchen92** [comment]
> @emkornfield I add a unit test and document, please take a look, thanks!
> And after rebasing code, build still fails with python test(have tried several times).

now build pass.

File: java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java (hunk 1 above)
Comment (praveenbingo) [reply java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java]: I am not sure following the Drill precedent is right, can we create a new method that explicitly calls out that this sizeOfValueBuffer alone?

Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java]: @praveenbingo Thanks for your comments, do you mean we just remove getCurrentSizeInBytes in VariableWidthVector and create a new method sizeOfValueBuffer instead? Not sure about that, what do you think? @emkornfield If you both think it is ok, I can update this PR, thanks!

**Comment by emkornfield** [comment]
I think it is ok to rename if you think it would be clearer because the old implementation did nothing

**Comment by praveenbingo** [comment]
> I think it is ok to rename if you think it would be clearer because the old implementation did nothing

Agreed, i think renaming to sizeOfValueBuffer() is more clear in meaning.

**Comment by tianchen92** [comment]
@emkornfield @praveenbingo Thanks a lot for your reply, I have updated this PR.

**Comment by wesm** [comment]
@tianchen92 note I've been editing PR titles to add spaces -- for some reason "[Java] Implement" is more readable than "[Java]Implement" to me

**Comment by emkornfield** [comment]
+1 thank you.

**Pull request closed without merging**